core/modules/field/modules/text/text.module

```diff
@@ -83,13 +103,31 @@ function text_field_instance_settings_form($field, $instance) {
       t('Plain text'),
       t('Filtered text (user selects text format)'),
     ),
+    '#weight' => 0,
   );
+  $form['allowed_formats'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('Allowed text formats'),
+    '#default_value' => $settings['allowed_formats'],
+    '#description' => t('Choose which <a href="@text_formats">text formats</a> should be available for authors. If an author does not have permission to a format on existing content, the field will be locked.', array('@text_formats' => url('admin/config/content/formats'))),
+    '#options' => $formats,
+    '#states' => array(
+      'visible' => array(
+        'input[name="instance[settings][text_processing]"]' => array('value' => 1),
+      ),
+    ),
+    '#weight' => 0,
+  );
+  $form['allowed_formats'][filter_fallback_format()]['#value'] = TRUE;
+  $form['allowed_formats'][filter_fallback_format()]['#disabled'] = TRUE;
+
   if ($field['type'] == 'text_with_summary') {
     $form['display_summary'] = array(
       '#type' => 'checkbox',
       '#title' => t('Summary input'),
       '#default_value' => $settings['display_summary'],
       '#description' => t('This allows authors to input an explicit summary, to be displayed instead of the automatically trimmed text when using the "Summary or trimmed" display type.'),
+      '#weight' => 0,
```

Comment:
This places this checkbox between the "Text processing" radios and the newly-introduced "Allowed text formats" checkboxes. Can we try either giving it a weight of `1`, or giving the other two a weight of `-1`?

Reply:
Thanks, good suggestion. I wasn't paying very close attention to the form element ordering. Fixed in latest commit.